server/sms/nexmo_sms_sender_test.go

```diff
@@ -35,3 +36,39 @@ func TestNexmoSender_SendWithError(t *testing.T) {
 	a.Error(err)
 	a.Equal(ErrRetryFailed, err)
 }
+
+func TestNexmoSender_SendExpiredMessage(t *testing.T) {
+	a := assert.New(t)
+
+	port := createRandomPort(7000, 8000)
+	URL = "http://127.0.0.1" + port
+
+	sender := createNexmoSender(t)
+	countCh := make(chan bool)
+	doneCh := make(chan struct{})
+	// no request should be made in case the sms is expired
+	go dummyNexmoEndpointWithHandlerFunc(t, countCh, port, func(t *testing.T, countCh chan bool) http.HandlerFunc {
```

Comment:
better way to do it.Make a new NexmoHandler and if a request reaches the http server.
a.FailNow() the request.In this way no doneCh is needed.